pkg/collector/corechecks/net/network/network.go

```diff
@@ -221,63 +427,275 @@ func submitProtocolMetrics(sender sender.Sender, protocolStats net.ProtoCounters
+func netstatAndSnmpCounters(procfsPath string, protocolNames []string) (map[string]net.ProtoCountersStat, error) {
+	counters := make(map[string]net.ProtoCountersStat)
+	for _, protocol := range protocolNames {
+		counters[protocol] = net.ProtoCountersStat{Protocol: protocol, Stats: make(map[string]int64)}
 	}
-	defer f.Close()
 
-	counters := map[string]int64{}
-	scanner := bufio.NewScanner(f)
-	for scanner.Scan() {
-		line := scanner.Text()
-		i := strings.IndexRune(line, ':')
-		if i == -1 {
-			return nil, errors.New("/proc/net/netstat is not fomatted correctly, expected ':'")
-		}
-		proto := strings.ToLower(line[:i])
-		if proto != "tcpext" {
-			continue
+	for _, subdirectory := range []string{"netstat", "snmp"} {
+		fs := filesystem
+		f, err := fs.Open(procfsPath + "/net/" + subdirectory)
+		if err != nil {
+			if subdirectory == "snmp" {
+				continue
+			}
+			return nil, err
 		}
+		defer f.Close()
+
+		scanner := bufio.NewScanner(f)
+		var protocolName string
+		for scanner.Scan() {
+			line := scanner.Text()
+			i := strings.IndexRune(line, ':')
+			if i == -1 {
+				return nil, errors.New(procfsPath + "/net/" + subdirectory + " is not fomatted correctly, expected ':'")
+			}
+			if contains(protocolNames, line[:i]) {
+				protocolName = line[:i]
+			} else {
+				continue
+			}
 
-		counterNames := strings.Split(line[i+2:], " ")
+			counterNames := strings.Split(line[i+2:], " ")
 
-		if !scanner.Scan() {
-			return nil, errors.New("/proc/net/netstat is not fomatted correctly, not data line")
-		}
-		line = scanner.Text()
+			if !scanner.Scan() {
+				return nil, errors.New(procfsPath + "/net/" + subdirectory + " is not fomatted correctly, not data line")
+			}
+			line = scanner.Text()
 
-		counterValues := strings.Split(line[i+2:], " ")
-		if len(counterNames) != len(counterValues) {
-			return nil, errors.New("/proc/net/netstat is not fomatted correctly, expected same number of columns")
+			counterValues := strings.Split(line[i+2:], " ")
+			if len(counterNames) != len(counterValues) {
+				return nil, errors.New(procfsPath + "/net/" + subdirectory + " is not fomatted correctly, expected same number of columns")
+			}
+			for j := range counterNames {
+				value, err := strconv.ParseInt(counterValues[j], 10, 64)
+				if err != nil {
+					return nil, err
+				}
+				counters[protocolName].Stats[counterNames[j]] = value
+			}
 		}
+	}
+	return counters, nil
+}
 
-		for j := range counterNames {
-			value, err := strconv.ParseInt(counterValues[j], 10, 64)
+func readIntFile(filePath string, fs afero.Fs) (int, error) {
+	data, err := afero.ReadFile(fs, filePath)
+	if err != nil {
+		return 0, err
+	}
+	value, err := strconv.Atoi(strings.TrimSpace(string(data)))
+	if err != nil {
+		return 0, err
+	}
+	return value, nil
+}
+
+func addConntrackStatsMetrics(sender sender.Sender, conntrackPath string, useSudoConntrack bool) {
+	cmd := []string{conntrackPath, "-S"}
+	if useSudoConntrack {
+		cmd = append([]string{"sudo"}, cmd...)
```

Comment:
question: I see this is from [the existing integration](https://github.com/DataDog/integrations-core/blob/master/network/datadog_checks/network/check_linux.py#L272-L274) but I really don't think we want to allow this, will discuss concerns further in slack